### PR TITLE
Team Colors

### PR DESCRIPTION
**Pull request by T00L00**
- ArenaManager randomly assigns colors to all teams in the scene
- Gatherer training scene was constantly resetting because there was only 1 team in the scene. Added a second gatherer team so Reset check doesn't fire off every frame. 
- Various bug fixes in prefab setups

### CODE REVIEW

File: Assets/Scripts/ArenaManager.cs [hunk 1]
```diff
@@ -72,13 +84,20 @@ private void EndEpisode()
...
                 teams[i].ResetTeam(i);
             }
             foodSpawner.ResetFood();
         }
+
+        private void SetTeamColors()
+        {
+            foreach (var team in teams)
+            {
+                team.SetTeamColor(UnityEngine.Random.ColorHSV(0f, 1f, 1f, 1f, 0.1f, 1f));
```
Comment (logar16) [inline Assets/Scripts/ArenaManager.cs:99]: Can we use something less random instead?  Something like... (you can tweak variables a little)

```csharp
        private Color NextColor(int count)
        {
            var shift = MAX_TEAMS / 2;
            float saturation = 1f;
            float value = 1f;
            if (count > (2 * MAX_TEAMS / 3))
            {
                shift += 1;
                saturation = 0.3f;
                value = 0.6f;
            }
            else if (count > (MAX_TEAMS / 3))
            {
                saturation = 0.5f;
            }
            count = (count * 5 + shift) % MAX_TEAMS;
            float hue = (float)count / MAX_TEAMS;
            return Color.HSVToRGB(hue, saturation, value);
        }
```

File: Assets/Scripts/RL/TeamGroup.cs [hunk 2]
```diff
@@ -43,6 +45,16 @@ public void ResetTeam(int teamId)
             storage.RemoveAllItems();
```
Comment (logar16) [inline Assets/Scripts/RL/TeamGroup.cs:45]: Not your problem, but can you reset the storage to be active in case it was destroyed?

**Review by logar16 — APPROVED** [review]
I've approved but you might consider changing some things I commented on

File: Assets/Scripts/ArenaManager.cs (hunk 1 above)
Comment (T00L00) [reply Assets/Scripts/ArenaManager.cs:99]: Will included in upcoming improvements

File: Assets/Scripts/RL/TeamGroup.cs (hunk 2 above)
Comment (T00L00) [reply Assets/Scripts/RL/TeamGroup.cs:45]: Will do

**Pull request merged**